oauthenticator/generic.py

```diff
@@ -84,10 +84,18 @@ class GenericOAuthenticator(OAuthenticator):
         help="Disable basic authentication for access token request",
     )
 
+    ca_certs = Unicode(
+        os.environ.get('OAUTH2_CA_CERTS', os.environ.get('SSL_CERT_FILE')),
+        allow_none=True,
+        config=True,
+        help="Custom trusted Certificate Authorities bundle in PEM format (ca_certs) or None to use defaults. Default: None",
```

Comment:
`or None to use defaults. Default: None",` is a bit confusing. I think you need to be explicit that "to use defaults" refer to fallback on the two separate environment variables you have mentioned - in sequence.

I'm not well versed in these environment variables. Are these common variable names used for these purposes etc? I think to approve this PR I need to learn more, and a key part would be to learn about these environment variables common meaning in other projects or defined meaning in this project.

Some questions:
- Does AsyncHTTPClient respect some environment variable already to configure this?
- Is there some RFC document declaring that these environment variable names are to be used for purposes like this, or is it up to anyone to invent names etc?

Thinking about this PR on a high level, allowing configuration of this makes sense, but defaulting to using environment variables that I don't know the background about etc could change the default behavior for users etc and requires more scrutiny.

Reply:
> `or None to use defaults. Default: None",` is a bit confusing. I think you need to be explicit that "to use defaults" refer to fallback on the two separate environment variables you have mentioned - in sequence.
> 
I modeled this help from the existing overrideable config via environment variable names, see lines 69-85, i.e. OAUTH2_USERDATA_REQUEST_TYPE, OAUTH2_TLS_VERIFY, OAUTH2_BASIC_AUTH, none of which calls out the environment variable names in the help. But I can adjust it to include those if you like and reword this as I do agree, that does sound a bit confusing.

Reply:
Do we need to support environment variables for this config property? Some of the `OAUTH2_*` env vars are due to the previous implementation of OAuthenticator where parameters couldn't be passed between components so are kept for backwards compatibility. Unless it's required I'm in favour of keeping things simple and not looking at env-vars for new properties.

Reply:
@consideRatio I just pushed update to help docs for new `ca_certs` config, please take a look if that is better.

@manics is there another way I can override the oauthenticate oauth2 ca-certs when I am running this from jupyterhub/k8s-hub image deployed to K8s via a helm chart? and I don't want to have to build another image from the k8s-hub image to inject my ca-certs into the system trusted CAs, e.g. I just want to override it via environment variables.

Reply:
Couldn't you set the property in the same way you set any other one, the Z2JH equivalent of `c.GenericOAuthenticator.ca_cert = ...`? Why is an environment variable also needed?